### PR TITLE
Rollout UI / server-side paging

### PR DESCRIPTION
**Pull request by tsagadar**
Some additional major changes

1. More flexibel swu parsing. Covers three different cases.
2. Rollout creation/deletion and pause/unpause addressing #27 
3. JavaScript formatting, modernisation and lint issue cleanup
4. Server-side filtering/search/paging

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
```
[the remark this answers is not in the document]
Comment (jameshilliard) [reply package.json:1]: I'm thinking maybe github actions would be best?

[the remark this answers is not in the document]
Comment (b-rowan) [reply package.json:1]: This gist seems to have what I want - https://gist.github.com/ahgraber/9ad4d0086a3f239f7872b7f33ebbe4c5#file-pre-commit-config-yaml-L59-L63

[the remark this answers is not in the document]
Comment (b-rowan) [reply package.json:1]: > I'm thinking maybe github actions would be best?

Not opposed to either, but I dont think github actions can make changes to files easily?  `pre-commit` is simplest for now.

[the remark this answers is not in the document]
Comment (jameshilliard) [reply package.json:1]: > Not opposed to either, but I dont think github actions can make changes to files easily? `pre-commit` is simplest for now.

Maybe [this](https://github.com/pre-commit-ci/lite-action)?

[the remark this answers is not in the document]
Comment (tsagadar) [reply package.json:1]: `mirrors-prettier` is archived stating _prettier made some changes that breaks plugins entirely_

[the remark this answers is not in the document]
Comment (tsagadar) [reply package.json:1]: If node is installed, then one can run the following command the clean-up files: `npx prettier . --write`

But ideally one has prettier support within the IDE. It is a bummer that pre-commit can no longer support it.

File: goosebit/ui/templates/devices.html [hunk 2]
```diff
@@ -1,100 +1,98 @@
...
+          data-bs-dismiss="modal"
+          onclick="updateDeviceName()"
+        >
+          Save changes
+        </button>
+      </div>
     </div>
+  </div>
 </div>
+<script src="{{ url_for('static', path='js/util.js') }}"></script>
```
[the remark this answers is not in the document]
Comment (tsagadar) [reply goosebit/ui/templates/devices.html]: Done.

File: goosebit/ui/static/js/index.js [hunk 3]
```diff
@@ -1,171 +1,191 @@
...
+      .querySelector("button.buttons-select-all")
+      .classList.add("disabled");
+  } else {
+    document
+      .querySelector("button.buttons-select-all")
+      .classList.remove("disabled");
+  }
 }
 
 function downloadLogins(devices) {
```
[the remark this answers is not in the document]
Comment (tsagadar) [reply goosebit/ui/static/js/index.js:153]: Done

File: package.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (b-rowan) [reply package.json:1]: Maybe Biome will work?  97% compatibility with prettier - https://github.com/biomejs/pre-commit

[the remark this answers is not in the document]
Comment (b-rowan) [reply package.json:1]:
```
  - repo: https://github.com/rbubley/mirrors-prettier
    rev: "v3.3.3"
    hooks:
      - id: prettier
```

This is working good for me, its using a different mirrors repo mentioned here - https://github.com/prettier/prettier/issues/15742

[the remark this answers is not in the document]
Comment (tsagadar) [reply package.json:1]: Switched to biome - also gives some linting besides formatting and has official support for pre-commit

**Comment by b-rowan** [comment]
Let me know when ready for review.  Biome doesnt seem to do HTML formatting (which is mildly annoying), but up to you which you prefer.  If we want to use prettier, see this comment - https://github.com/UpstreamDataInc/goosebit/pull/37#discussion_r1697274155

File: package.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (b-rowan) [reply package.json:1]: > Switched to biome - also gives some linting besides formatting and has official support for pre-commit

Ok.  Should no longer need `package.json` then, as pre-commit handles its own envs.

[the remark this answers is not in the document]
Comment (tsagadar) [reply package.json]: Correct. Removed and fixed commit message. I would give biomejs a chance - sounds like pre-commit & prettier don't have a great future. Not having HTML formatting is the obvious drawback.

[the remark this answers is not in the document]
Comment (b-rowan) [reply package.json]: > Correct. Removed and fixed commit message. I would give biomejs a chance - sounds like pre-commit & prettier don't have a great future. Not having HTML formatting is the obvious drawback.

Is it a stupid idea to just use prettier and have it only include HTML files?

**Comment by tsagadar** [comment]
> Let me know when ready for review. Biome doesnt seem to do HTML formatting (which is mildly annoying), but up to you which you prefer. If we want to use prettier, see this comment - [#37 (comment)](https://github.com/UpstreamDataInc/goosebit/pull/37#discussion_r1697274155)

@b-rowan Ready.

**Review by b-rowan — APPROVED** [review]

**Pull request merged**